python/cdp_missoula_backend/scraper.py

```diff
@@ -69,38 +88,95 @@ def get_scraped_data() -> List:
             for meeting in meetings:
                 if meeting.is_displayed():
                     _meeting_data = {}
-                    title = meeting.find_element(
-                        by=By.XPATH, value=".//div[@class='meeting-title']"
-                    )
-                    _meeting_data["title"] = title.text
 
-                    date = meeting.find_element(
+                    meeting_date = meeting.find_element(
                         by=By.XPATH, value=".//div[@class='meeting-date']"
                     )
-                    # TODO: Convert to datetime
-                    _meeting_data["date"] = date.text
-                    
+                    # Thursday, 3 February 2022 @ 10:00 AM
+                    converted_dt = datetime.strptime(
+                        meeting_date.text, "%A, %d %B %Y @ %I:%M %p"
+                    )
+                    if from_dt <= converted_dt <= to_dt:
+                        _meeting_data["date"] = converted_dt
+                    else:
+                        continue
+
                     try:
                         video = meeting.find_element(
                             by=By.XPATH,
                             value=".//li[@class='resource-link']//a[contains(@href, '/Players/ISIStandAlonePlayer.aspx?')]",
                         )
                     except NoSuchElementException:
-                        pass
+                        # if there is no video for the meeting, don't include it for rendering
+                        continue
                     else:
                         _meeting_data["video_player_uri"] = video.get_attribute("href")
 
+                    title = meeting.find_element(
+                        by=By.XPATH, value=".//div[@class='meeting-title']"
+                    )
+                    _meeting_data["title"] = title.text
+
                     meetings_info.append(_meeting_data)
 
     # navigate to each video player and construct the direct .mp4 uri
     for info in meetings_info:
         if "video_player_uri" in info:
             driver.get(info["video_player_uri"])
+            # TODO: Make duration calculation optional and have this look for isi_player instead
+            try:
+                WebDriverWait(driver, 10).until(
+                    EC.visibility_of_element_located(
+                        (
+                            By.XPATH,
+                            "//span[@class='fp-duration']",
+                        )
+                    )
+                )
+            except:
+                # it's likely this video is corrupt, but give peace a chance to see if
+                # the file_name is present
+                pass
+
             player = driver.find_element(by=By.XPATH, value="//div[@id='isi_player']")
             file_name = player.get_attribute("data-file_name")
+            if file_name.count(".mp4") == 0:
+                print(f"Erroring out on file_name {file_name}")
```

Comment:
It would be fun to auto-email Missoula City and let them know their video link is broken

Reply:
```
Erroring out on file_name live2
{'date': datetime.datetime(2022, 2, 15, 16, 0), 'video_player_uri': 'https://pub-missoula.escribemeetings.com/Players/ISIStandAlonePlayer.aspx?Id=ecf3882b-25dd-4d9b-9b1c-2ca993e94a10', 'title': 'Public Art Committee'}
Erroring out on file_name live2
{'date': datetime.datetime(2022, 1, 18, 16, 0), 'video_player_uri': 'https://pub-missoula.escribemeetings.com/Players/ISIStandAlonePlayer.aspx?Id=9b84a1c5-81a3-4999-8150-219159b2a1b2', 'title': 'Public Art Committee'}
```